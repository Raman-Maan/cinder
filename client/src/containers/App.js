import React, { Component } from 'react';
import logo from '../images/logo.svg';
import './App.css';

class App extends Component {
  state = {
    server_response: ''
  };

  getMessage = () => {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  };

  componentDidMount() {
    this.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
