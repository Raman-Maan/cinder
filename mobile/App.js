import React from 'react';
import Expo, { AppLoading } from 'expo';

import Home from './containers/home';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fontLoaded: false,
    };
  }

  componentDidMount() {
    this.loadAssets();
  }

  async loadAssets() {
    await Expo.Font.loadAsync({
      Roboto: require('native-base/Fonts/Roboto.ttf'),
      Roboto_medium: require('native-base/Fonts/Roboto_medium.ttf'),
    });

    this.setState({ fontLoaded: true });
  }

  render() {
    if (!this.state.fontLoaded) {
      return <AppLoading />;
    }

    return (
      <Home />
    );
  }
}
