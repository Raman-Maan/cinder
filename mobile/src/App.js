import React from 'react';
import { View, StatusBar } from 'react-native';
import { StackNavigator } from 'react-navigation';

import Home from './containers/home';
import Profile from './containers/blank';
import Matches from './containers/matches';

const RootStack = StackNavigator({
  Home: {
    screen: Home,
  },
  Profile: {
    screen: Profile,
  },
  Matches: {
    screen: Matches,
  },
}, {
  initialRouteName: 'Matches',
  navigationOptions: {
    title: 'cinder',
  },
});

export default class App extends React.Component {
  render() {
    return (
      <RootStack />
    );
  }
}
