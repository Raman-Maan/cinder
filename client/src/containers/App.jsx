import React, { Component } from 'react';
import logo from '../assets/logo.svg';
import './App.css';

class App extends Component {
  static getMessage() {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  }

  constructor() {
    super();
    this.state = {
      server_response: 'No request for message has been made'
    };
  }

  componentDidMount() {
    App.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
