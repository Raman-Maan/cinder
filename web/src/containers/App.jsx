import React, { Component } from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import PrivateRoute from '../components/PrivateRoute';
import './styles/App.css';
import { connect } from 'react-redux';

import Layout from './Layout';
import Login from './Login';
import Signup from './Signup';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Switch>
          <Route exact path="/login" component={Login} />
          <PrivateRoute
            exact
            path="/"
            authed={this.props.isAuthenticated}
            component={Layout}
          />
          <Route exact path="/signup" component={Signup} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default withRouter(connect(mapStateToProps, null)(App));
