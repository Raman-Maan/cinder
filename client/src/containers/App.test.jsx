import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import App from './App';

describe('<App />', () => {
  it('renders without crashing', () => {
    // simple smoke test
    shallow(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
  });

  it('renders correctly', () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <App />
        </MemoryRouter>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
