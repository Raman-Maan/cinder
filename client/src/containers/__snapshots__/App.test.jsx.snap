// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="HomeContent"
  >
    <div
      className="NavigationBar"
    >
      <nav
        className="navbar bg-faded"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          cinder
        </a>
        <ul
          className="navItem navbar-nav"
        >
          <li
            className="nav-item"
          >
            <img
              alt="User's Portrait"
              className="UserImage"
              src="https://mixtapemonkey.com/artistpic/278.jpg"
            />
          </li>
          <li
            className="nav-item"
          >
            <h5
              className="UserName"
            >
              Ugly God
            </h5>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="Matches"
    >
      <div
        className="container-fluid"
      >
        <div
          className="full row"
        >
          <div
            className="col col-sm-4"
          >
            <div
              className="MatchesList"
            >
              <div
                className="empty"
              >
                <h3
                  className="msg align-self-center"
                >
                  No matches yet. Get swiping!
                </h3>
              </div>
            </div>
          </div>
          <div
            className="col col-sm-8"
          >
            <div
              className="UserDetail"
            >
              <div
                className="empty"
              >
                <h3
                  className="msg align-self-center"
                >
                  No match selected yet. Select one match from the list!
                </h3>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
